packages/stats/gbstats/bayesian/tests.py

```diff
@@ -223,3 +240,129 @@ def get_risk(mu, sigma) -> List[float]:
         risk_ctrl = float((1.0 - prob_ctrl_is_better) * mn_pos)
         risk_trt = -float(prob_ctrl_is_better * mn_neg)
         return [risk_ctrl, risk_trt]
+
+
+class Bandits(object):
+    def __init__(
+        self,
+        stats: Sequence[
+            Union[ProportionStatistic, SampleMeanStatistic, RatioStatistic]
+        ],
+        config: BanditConfig,
+    ):
+        self.stats = stats
+        self.config = config
+        self.n_variations = len(stats)
+
+    @property
+    def variation_means(self) -> np.ndarray:
+        return np.array([arm.mean for arm in self.stats])
+
+    @property
+    def variation_variances(self) -> np.ndarray:
+        return np.array([arm.variance for arm in self.stats])
+
+    @property
+    def variation_counts(self) -> np.ndarray:
+        return np.array([arm.n for arm in self.stats])
+
+    @property
+    def prior_precision(self) -> np.ndarray:
+        return np.full(
+            (self.n_variations,),
+            int(self.config.prior_distribution.proper)
+            / self.config.prior_distribution.variance,
+        )
+
+    @property
+    def data_precision(self) -> np.ndarray:
+        return np.array(
+            [
+                float(n) / v if v > 0 else 0
+                for n, v in zip(self.variation_counts, self.variation_variances)
+            ]
+        )
+
+    @property
+    def posterior_precision(self) -> np.ndarray:
+        return self.prior_precision + self.data_precision
+
+    @property
+    def posterior_variance(self) -> np.ndarray:
+        return 1 / self.posterior_precision
+
+    @property
+    def prior_mean(self) -> np.ndarray:
+        return np.full((self.n_variations,), self.config.prior_distribution.mean)
+
+    @property
+    def posterior_mean(self) -> np.ndarray:
+        return self.posterior_variance * (
+            self.prior_precision * self.prior_mean
+            + self.data_precision * self.variation_means
+        )
+
+    # number of Monte Carlo samples to perform when sampling to estimate weights for the SDK
+    @property
+    def n_samples(self):
+        return int(1e4)
+
+    # function that computes thompson sampling variation weights
+    def compute_variation_weights(self) -> BanditWeights:
+        min_n = 100
+        if any(self.variation_counts < min_n):
+            update_message = "some variation counts smaller than " + str(min_n)
+            p = np.full((self.n_variations,), 1 / self.n_variations).tolist()
+            return BanditWeights(update_message=update_message, weights=p)
+        y = np.random.multivariate_normal(
```

Comment:
The numpy docs says to use https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.multivariate_normal.html#numpy.random.Generator.multivariate_normal instead

Reply:
updated, thx